doc/coq.rst

```diff
@@ -0,0 +1,383 @@
+.. _coq:
+
+***
+Coq
+***
+
+.. contents:: Table of Contents
+    :depth: 3
+
+Introduction
+------------
+
+Dune is able to build Coq theories and plugins with support for extraction and
+.mlg file preprocessing.
+
+A *Coq theory* is a collection of ``.v`` files, perhaps in several
+subdirectories, all prefixed by a common Coq module name, following the
+directory hierarchy for module names.
+
+A *Coq plugin* is an OCaml :ref:`library`, that is intended to be dynamically
+loaded by Coq at runtime when compiling a Coq theory, and is typically linked
+with the Coq OCaml API.
+
+We may indistinctively refer to a :ref:`dune-project` containing Coq theories as
+a *Coq project*.
+
+A *Coq project* is a collection of Coq plugins and Coq theories. This notion is
+subsumed by that of a :ref:`dune-project`, and should not be considered a
+technical definition.
+
+The ``.v`` files of a theory do not only have to be present as source files.
+They may also be created as Dune targets from other rules.
+
+.. _coq-theory:
+
+coq.theory
+----------
+
+To enable Coq support in a Dune project. The Coq language version should be
```

Comment:
```suggestion
Enables Coq support in a Dune project. The Coq language version should be
```

Reply:
I will reorder the information here. coq.theory is not being discussed immediately, but instead what you need to include in your dune project in order to enable its use. It is a sort of mini introduction.

I've separated these parts so that it reads more clearly.